query/src/main/scala/filodb/query/exec/MultiSchemaPartitionsExec.scala

```diff
@@ -43,14 +44,15 @@ final case class MultiSchemaPartitionsExec(id: String,
 
     // Find the schema if one wasn't supplied
     val schemas = source.schemas(dataset).get
-    val dataSchema = schema.map { s => schemas.schemas(s) }
-                           .getOrElse(schemas(
-                             lookupRes.firstSchemaId.getOrElse(schemas.schemas.values.head.schemaHash)))
 
     // If we cannot find a schema, or none is provided, we cannot move ahead with specific SRPE planning
+    // Also warn if unknown schema is found.  This is not supposed to happen since ingestion filters out unknown schemas
     schema.map { s => schemas.schemas(s) }
           .orElse(lookupRes.firstSchemaId.map(schemas.apply))
-          .map { sch =>
+          .filter { sch =>
+            if (sch == Schemas.UnknownSchema) qLogger.warn(s"Unknown schema ID ${lookupRes.firstSchemaId} skipped")
```

Comment:
Do we really need this as a warn? Won't it flood the logs for each query when the issue occurs?

Reply:
Should be rare.  I can downgrade it to an info though.

Reply:
I think we should leave it as WARN.   This only happens when we switch schemas and tables have not been truncated.  The WARNing should help alert users that that is what's going on.

Actually, let's throw a specific error so that operators can figure out what's going on easily.

Reply:
ok fixed.